tests/core/non_linear_test_system_01.h

```diff
@@ -17,18 +17,8 @@
  * Author: Simon Gauvin, Polytechnique Montreal, 2019
  */
 
-#include <deal.II/lac/full_matrix.h>
-#include <deal.II/lac/lapack_full_matrix.h>
-#include <deal.II/lac/sparse_direct.h>
-#include <deal.II/lac/vector.h>
-
-#include <core/newton_non_linear_solver.h>
-#include <core/non_linear_solver.h>
 #include <core/physics_solver.h>
 
-#include <iostream>
-#include <memory>
-
```

Comment:
Are you sure? To me it seems like the .h would not compile without these includes. Or it makes it depend from the includes of another file?